### PR TITLE
Updated revert WDL to split RGs up front for scalability

### PR DESCRIPTION
**Pull request by vdauwera**
Splitting by RG up front makes RevertSam far more scalable (in part by avoiding a huge sort step) and allows us to scatter the reversion.

### CODE REVIEW

**Comment by vdauwera** [comment]
Small-scale test [c929d447-96b0-4f71-8124-12ed3bf180ef](https://cromwell-v24.dsde-methods.broadinstitute.org/api/workflows/v1/c929d447-96b0-4f71-8124-12ed3bf180ef/status)

**Comment by vdauwera** [comment]
SplitByRG works great but a weird error occurs when running on the output RG bams:

````
Exception in thread "main" htsjdk.samtools.SAMException: Cannot determine candidate qualities: no qualities found.
	at htsjdk.samtools.util.QualityEncodingDetector.generateCandidateQualities(QualityEncodingDetector.java:228)
	at htsjdk.samtools.util.QualityEncodingDetector.generateBestGuess(QualityEncodingDetector.java:379)
	at htsjdk.samtools.util.QualityEncodingDetector.detect(QualityEncodingDetector.java:332)
	at picard.sam.RevertSam.createReadGroupFormatMap(RevertSam.java:528)
	at picard.sam.RevertSam.doWork(RevertSam.java:265)
	at picard.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:205)
	at picard.cmdline.PicardCommandLine.instanceMain(PicardCommandLine.java:94)
	at picard.cmdline.PicardCommandLine.main(PicardCommandLine.java:104)
````

Validating the RG files now at [b7b927df-2d05-4d4f-95c3-957208925734](https://cromwell-v24.dsde-methods.broadinstitute.org/api/workflows/v1/b7b927df-2d05-4d4f-95c3-957208925734/status).

File: scripts/broad_dsde_workflows/RevertBamToUnmappedRGBamsWf_170306.wdl [hunk 1]
```diff
@@ -26,33 +26,59 @@
...
   }
   runtime {
-    docker: "broadinstitute/genomes-in-the-cloud:2.2.3-1469027018"
+    docker: "broadinstitute/genomes-in-the-cloud:2.2.5-1486412288"
     disks: "local-disk " + disk_size + " HDD"
     memory: mem_size
   }
   output {
-    Array[File] unmapped_bams = glob("*.bam")
+    File unmapped_bam = "${output_name}"
```
Comment (cjllanwarne) [inline scripts/broad_dsde_workflows/RevertBamToUnmappedRGBamsWf_170306.wdl]: This could (probably should) just be `File unmapped_bam = "output_name"`

**Comment by vdauwera** [comment]
Better solution in https://github.com/broadinstitute/wdl/pull/138

**Pull request closed without merging**